core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -912,13 +914,39 @@ private[spark] class TaskSchedulerImpl(
     }
   }
 
-  override def executorDecommission(executorId: String): Unit = {
+  override def executorDecommission(
+      executorId: String, decommissionInfo: ExecutorDecommissionInfo): Unit = {
+    synchronized {
```

Comment:
Is there any guarantee that a `ExecutorDecommissionInfo` with `isHostDecommissioned=false` is never followed by another `ExecutorDecommissionInfo` where `isHostDecommissioned=true`?

Reply:
That would be a bug: isHostDecommissioned is ultimately decided by the cluster manager. It knows for sure whether or not the host is decommissioned with the executor. It shouldn't change its mind :-). I can add an assert in the Driver but it would be a bit of an overkill.

Reply:
I have changed this to putIfAbsent

Reply:
Changed this to "last writer wins"

Reply:
So my concern here is that if a task is scheduled part of the way through decommissioning the executor *could* send back a decommission message with the host lost. Maybe use a getOrElse to make sure if we've ever learnt the entire host is going away we don't forget that?

Reply:
Ah, I see your point -- its because of the DecommissionExecutor messages that the executor sends to the driver when work is assigned to it. And there, it can reply back with a hostDecommissioned = false, because the executor itself does not know if the host is going away or not (only the cluster manager knows). 

I like your suggestion of doing this "merging" semantics, it is easier/less-work than plumbing host Lost to the executor :-)